### DISCUSSION TITLE
Not sure how to get started

### DISCUSSION THREAD

**Discussion by ghost** [TSnake41/raylib-lua#8] [Q&A]
Hi, I'd love to try Raylib with Lua, but I'm honestly not sure how to get started. Is there a more detailed explanation/step-by-step guide what to do in order to get started to run a first example project? I'm on macOS, but Windows would also be possible in case there are no resources for macOS.

Any help appreciated! Thank you!

**Comment by aum7** [comment]
olo

to get started, one needs to reverse readme.md instructions (i am using mint linux, windows are similar) :
- get / dl / clone raylib-lua code, ie in terminal (cmd if on windows), navigate to directory where you want raylib-lua to be, cca 500 mb of space is needed : 
`$ cd vmdev`
![230307 raylib-terminal](https://user-images.githubusercontent.com/9611505/223418392-ac2bfbeb-aaad-4038-9eb9-a2696752d697.png)

- clone github repository / code, git needs to be installed on system : 
`vmdev$ git clone https://github.com/TSnake41/raylib-lua.git`
- this will copy all needed code to vmdev/raylib-lua directory; navigate into cloned directory : 
`vmdev$  cd raylib-lua`
- also install submodules : 
`vmdev/raylib-lua$ git submodule init`
`vmdev/raylib-lua$ git submodule update`
- build / compile with make, which also must be installed on system : 
`vmdev/raylib-lua$ make`
- this will produce some code in the terminal (cmd), and probably some (or lot) of compile warnings - they can be ignored, raylib-lua should work just fine
- we are ready to go; let us run an included example from examples folder (raylib_e & raylib_s are in a main raylib-lua folder) : 
`vmdev/raylib-lua$ ./raylua_s examples/core_basic_window.lua`
- this will run basic window example, from terminal - because we used raylua_s(cript) command to run an example
- exit the basic window app with 'esc' key, or click 'close [x]' window icon
- if we run : 
`vmdev/raylib-lua$ ./raylib_e examples/core_basic_window.lua`
- the raylib_e(mbed) is smart enough to make executable (stand-alone, not run in terminal) file, in examples folder, called 'core_basic_window_out' (.exe, i guess, on windows); we can run this file, by typing : 
`vmdev/raylib-lua$  ./examples/core_basic_window_out`
- or (same result) : 
`vmdev/raylib-lua$ examples/core_basic_window_out`
- and we get exactly the same window as previously; hit 'esc' or click [x] 'close window' to exit your fresh-baked, still warm application
- we can also run the application by double-clicking its name, in 'file explorer' (on linux, it must have execution rights, which, in this case, by default it does)

![230307 raylib basic-window](https://user-images.githubusercontent.com/9611505/223417057-107cac85-9999-48dd-94da-675219b2e451.png)

now open some more examples, put the code in the tin can, or glass jar, close tightly, and shake the can / jar, until the code promises that it will behave well, and run nicely. lol
ps : if code stubbornly refuses to behave / run nicely, threaten it with boiling water - this should convince it to run nice. lol 100x

all hail TSnake41 !
big thanks to master tsnake41, for sharing such exemplary piece of code
bow !

have fun

aum
s love nia

**Comment by originalbluesin** [comment]
I am so sorry to dig this old thread up. I am following the step and met with error at 'make'. If there is a chance you are seeing this, please help me. I really want to try Raylib but i don't know any other programming language but Lua. (Windows 10 environment.)
![make](https://github.com/TSnake41/raylib-lua/assets/91607043/8a285548-17a8-47b0-a325-42e9c4ca7c47)

Thank you.

* edited. I have found the complied and build exe. So i skip this build process and use it directly. Sorry for your inconvenient.
